core/src/main/scala/kafka/controller/ReplicaStateMachine.scala

```diff
@@ -190,15 +189,14 @@ class ReplicaStateMachine(controller: KafkaController) extends Logging {
                                                 targetState))
             case _ =>
               // check if the leader for this partition ever existed
-              controllerContext.partitionLeadershipInfo.get(topicAndPartition) match {
-                case Some(leaderIsrAndControllerEpoch) =>
-                  brokerRequestBatch.addLeaderAndIsrRequestForBrokers(List(replicaId), topic, partition, leaderIsrAndControllerEpoch,
-                    replicaAssignment)
-                  replicaState.put(partitionAndReplica, OnlineReplica)
-                  stateChangeLogger.trace("Controller %d epoch %d changed state of replica %d for partition %s from %s to %s"
-                    .format(controllerId, controller.epoch, replicaId, topicAndPartition, currState, targetState))
-                case None => // that means the partition was never in OnlinePartition state, this means the broker never
-                  // started a log for that partition and does not have a high watermark value for this partition
+              // None means the partition was never in OnlinePartition state, this means the broker never
```

Comment:
@hachikuji maybe you are right:) I am going to refactor those parts of the code. Thanks for the review.